lib/stacks/eks-blueprint-stack.ts

```diff
@@ -38,23 +39,106 @@ export class EksBlueprintProps {
     /**
      * Kubernetes version (must be initialized for addons to work properly)
      */
-    readonly version?: KubernetesVersion = KubernetesVersion.V1_19;
+    readonly version?: KubernetesVersion = KubernetesVersion.V1_20;
 
     /**
      * VPC
      */
     readonly vpc?: Vpc;
 }
 
+
+/**
+ * Blueprint builder implements a builder pattern that improves readability (no bloated constructors)
+ * and allows creating a blueprint in an abstract state that can be applied to various instantiations 
+ * in accounts and regions. 
+ */
+export class BlueprintBuilder implements AsyncStackBuilder {
+
+    private props: Partial<EksBlueprintProps>;
+    private env: {
+        account?: string,
+        region?: string
+    }
+
+    constructor() {
+        this.props = { addOns: new Array<ClusterAddOn>(), teams: new Array<Team>()};
+        this.env = {};
+    }
+
+    public name(name: string) : this {
```

Comment:
same thing here...this pattern seems odd to me. Also returning this? Is that valid typescript?

Reply:
I was using this for years with different programming languages. You can try `make build` `make run-test` `make lint` on the branch.